beep/protocol_tools/maccor_to_arbin.py

```diff
@@ -0,0 +1,407 @@
+                limit['Equation0_szCompareSign'] = end['Oper'].replace(' ', '')
+            elif step_type == 'Dischrge':
+                limit['Equation0_szRight'] = '-' + end['Value']
+                limit['Equation0_szCompareSign'] = end['Oper'].replace(' ', '').replace('<', '>')
+            else:
+                raise ValueError("Unable to convert end to limit for EndType:{} and Ctrl:{}".
+                                 format(end['EndType'], blank_step['m_szStepCtrlType']))
+        elif end['EndType'] == 'Current ' and blank_step['m_szStepCtrlType'] == "Voltage(V)":
+            limit['Equation0_szLeft'] = 'PV_CHAN_Current'
+            if step_type == ' Charge ':
+                limit['Equation0_szRight'] = end['Value']
+                limit['Equation0_szCompareSign'] = end['Oper'].replace(' ', '')
+            elif step_type == 'Dischrge':
+                limit['Equation0_szRight'] = '-' + end['Value']
+                limit['Equation0_szCompareSign'] = end['Oper'].replace(' ', '').replace('<', '>')
+            else:
+                raise ValueError("Unable to convert end to limit for EndType:{} and Ctrl:{}".
+                                 format(end['EndType'], blank_step['m_szStepCtrlType']))
+        elif end['EndType'] == 'StepTime':
+            limit['Equation0_szLeft'] = 'PV_CHAN_Step_Time'
+            limit['Equation0_szCompareSign'] = '>'
+            if '.' in end['Value']:
+                nofrag, frag = end['Value'].split(".")
+                frag = frag[:6]  # truncate to microseconds
+                frag += (6 - len(frag)) * '0'  # add 0s
+                elapsed = datetime.strptime(nofrag.replace('::', '00:00:0'),
+                                            "%H:%M:%S").replace(microsecond=int(frag)) - \
+                    datetime.strptime("00:00:00", "%H:%M:%S")
+            else:
+                elapsed = datetime.strptime(end['Value'].replace('::', '00:00:0'), "%H:%M:%S") - \
+                    datetime.strptime("00:00:00", "%H:%M:%S")
+            limit['Equation0_szRight'] = str(elapsed.total_seconds())
+        elif end['EndType'] == 'Loop Cnt':
+            loop_counter = int(re.search(r'\d+', step_type).group())
+            limit['Equation0_szLeft'] = 'TC_Counter{}'.format(loop_counter)
+            limit['Equation0_szCompareSign'] = end['Oper'].replace(' ', '')
+            limit['Equation0_szRight'] = end['Value']
+
+        elif end['EndType'] == 'Loop Addendum':
+            loop_counter = int(re.search(r'\d+', step_type).group())
+            limit['m_szGotoStep'] = step_flow_ctrl[step_index]
+            limit['Equation0_szLeft'] = 'TC_Counter{}'.format(loop_counter)
+            limit['Equation0_szCompareSign'] = '<'
+            limit['Equation0_szRight'] = end['Value']
+
+        else:
+            raise ValueError("Unable to set end for type {}".format(end['EndType']))
+
+        return limit
+
+    def convert_report_to_limit(self, report):
```

Comment:
Maybe rename this method (limits for logging) to `convert_reporting_to_logging_limits` to disambiguate from the previous method (limits for step termination)?

Reply:
Good idea, will do that!